### ISSUE TITLE
Mypy doesn't understand structural subtyping for nested TypedDicts

### ISSUE THREAD

**Issue by sh-at-cs** [python/mypy#17754]
**Bug Report**

When a structural subtyping relation between two "inner" `TypedDict`s should cause "outer" `TypedDict`s that contain them (and only them) to be subtypes of each other as well, Mypy fails to understand this and reports that the outer subtype is incompatible with the outer supertype.

**To Reproduce**

```python
from typing import TypedDict

class InnerSuper(TypedDict):
  x: str

class InnerSub(TypedDict):
  x: str
  y: int

class OuterSuper(TypedDict):
  inner: InnerSuper

class OuterSub(TypedDict):
  inner: InnerSub

# Mypy understands that InnerSub is a structural subtype of InnerSuper...
inner_sub: InnerSub
inner_super: InnerSuper = inner_sub  # no type error

# ... but not that OuterSub is a structural subtype of OuterSuper:
outer_sub: OuterSub
outer_super: OuterSuper = outer_sub  # type error
```

**Expected Behavior**

No type errors are reported by Mypy because `OuterSub` is a structural subtype of `OuterSuper`.

**Actual Behavior**

Mypy reports an error:

```
example.py:22: error: Incompatible types in assignment (expression has type "OuterSub", variable has type "OuterSuper")  [assignment]
Found 1 error in 1 file (checked 1 source file)
```

**Your Environment**

- Mypy version used: 1.11.2
- Mypy command-line flags: None
- Mypy configuration options from `mypy.ini` (and other config files): None
- Python version used: 3.10.12

**Comment by JelleZijlstra** [comment]
This is correct because TypedDict is invariant in its value types. It would be legal to write `outer_super["inner"] = {"x": 3}`, making the value incompatible with `InnerSub`.

https://peps.python.org/pep-0705/ will address this.
